File: templates/dayX.py
import sys


def parse_input(filename):
    lines = []
    with open(filename, "r") as f:
        lines = list(map(lambda s: s.replace("\n", ""), f.readlines()))
    return lines


def part1(lines):
    pass


def part2(lines):
    pass


def main():
    lines = parse_input(sys.argv[1])
    part1(lines)
    part2(lines)


if __name__ == "__main__":
    main()

File: templates/dayX.rs
use std::{env, fs::read_to_string};

fn parse_input(filename: &str) -> Vec<String> {
    read_to_string(filename)
        .unwrap()
        .lines()
        .map(String::from)
        .collect()
}

fn part1(lines: &Vec<String>) {}

fn part2(lines: &Vec<String>) {}

pub fn main() {
    let args: Vec<String> = env::args().collect();
    let lines = parse_input(&args[1]);
    part1(&lines);
    part2(&lines);
}
